evm/src/main/java/org/hyperledger/besu/evm/internal/FlexStack.java

```diff
@@ -163,12 +181,21 @@ public void push(final T operand) {
       throw new OverflowException();
     }
     if (nextTop >= currentCapacity) {
-      expandEntries(Math.min(currentCapacity + INCREMENT, maxSize));
+      final int newCapacity = newLength(currentCapacity, currentCapacity + 1, currentCapacity >> 1);
+      expandEntries(Math.min(newCapacity, maxSize));
     }
     entries[nextTop] = operand;
     top = nextTop;
   }
 
+  private int newLength(final int oldCapacity, final int minGrowth, final int prefGrowth) {
+    final int growth = Math.max(minGrowth, prefGrowth);
```

Comment:
from line 184
      final int newCapacity = newLength(currentCapacity, currentCapacity + 1, currentCapacity >> 1);
won't `growth` always end up being currentCapacity + 1 ?
currentCapacity >> 1 gives 0.5 x currentCapacity which will be less?